### ISSUE TITLE
Authentication failures Hashicorp Vault on Kubernetes 1.21

### ISSUE THREAD

**Issue by sjoel** [hashicorp/vault#13867]
**Describe the bug**
ARO (Azure Redhat Openshift) cluster is upgraded from 4.7 to 4.8 version. However, we see issue with pods stuck in "init 0/1" state indicating connection errors.
Error from server (BadRequest): container "anm" in pod "anm-xxx-xxx" is waiting to start: PodInitializing

* claim "iss" is invalid" backoff=2.160920171
2022-02-01T09:29:24.556Z [INFO] auth.handler: authenticating
2022-02-01T09:29:24.564Z [ERROR] auth.handler: error authenticating: error="Error making API request.

Vault pod logs show:
[INFO] expiration: revoked lease:
lease_id=auth/kubernetes/login

**To Reproduce**
1. Run Hashicorp on Kubernetes 1.21 
We are referring to articles that say when kubernetes is upgraded to >= 1.21, there are issues with the vault and it needs a repair https://particule.io/en/blog/vault-1.21/

**Expected behavior**
All application pods should be running instead they are stuck in init 0/1 state after the kubernetes (ARO) cluster upgrade

**Environment:**
Vault Server Version : HA Mode  active
Vault CLI Version : 1.42
Server Operating System/Architecture: Vault Pods deployed in Openshift

Vault server configuration file(s):
Will share the vault yaml config files to the Developer once issue is looked at. 

Please suggest on the next actions.

**Comment by RemcoBuddelmeijer** [comment]
Depending on whether you work with an internal or external Vault cluster either refer to:
1. (If working with an internal cluster) The [documentation](https://www.vaultproject.io/docs/auth/kubernetes#kubernetes-1-21) for working with the newly introduced [bound service account token volume](https://kubernetes.io/docs/reference/access-authn-authz/service-accounts-admin/#bound-service-account-token-volume) (introduced through beta in Kubernetes 1.21). 
2. (If working with an external cluster) The option `disable_iss_validation` to disable local issuer validation. This option is enabled by default since this [commit](https://github.com/hashicorp/vault-plugin-auth-kubernetes/commit/e2a3bbe7fa1c3403229415e007e852c4b2962da8#diff-c6f0305d05f9581c583eb2b15d17d32f412dcbc73809741b065ab801cc5d360e) and deprecated since [v0.7.1](https://github.com/hashicorp/vault-plugin-auth-kubernetes/releases/tag/v0.7.1) of the kubernetes auth plugin.

**Comment by sjoel** [comment]
Thanks Remco for the above info.. This helped us a lot!!

We were able to fix the issue and did not downgrade our cluster.
Followed the guide that you have shared - https://www.vaultproject.io/docs/auth/kubernetes#kubernetes-1-21
Added the additional config to Vault using the vault CLI

vault write auth/kubernetes/config \
     issuers="<your issuer>" \
    token_reviewer_jwt="<your reviewer service account JWT>" \
    kubernetes_host=https://IP:<your TCP port or blank for 443> \
    kubernetes_ca_cert=@ca.crt \
    disable_iss_validation=true

And verified if issuer is updated
vault read auth/kubernetes/config 

Tada! It worked! 

Thanks again for the quick response.
